x/evm/types/denom_config_testing.go

```diff
@@ -31,8 +34,30 @@ const (
 // EvmCoinInfo struct holds the name and decimals of the EVM denom. The EVM denom
 // is the token used to pay fees in the EVM.
 type EvmCoinInfo struct {
-	Denom    string
-	Decimals Decimals
+	Denom        string
+	DisplayDenom string
+	Decimals     Decimals
+}
+
+// ChainsCoinInfo is a map of the chain id and its corresponding EvmCoinInfo
+// that allows initializing the app with different coin info based on the
+// chain id
+var ChainsCoinInfo = map[string]EvmCoinInfo{
+	utils.MainnetChainID: {
+		Denom:        types.BaseDenom,
+		DisplayDenom: types.DisplayDenom,
+		Decimals:     EighteenDecimals,
+	},
+	utils.TestnetChainID: {
+		Denom:        types.BaseDenomTestnet,
+		DisplayDenom: types.DisplayDenomTestnet,
+		Decimals:     EighteenDecimals,
+	},
+	utils.SixDecChainID: {
+		Denom:        types.BaseDenom,
+		DisplayDenom: types.DisplayDenom,
+		Decimals:     SixDecimals,
+	},
```

Comment:
Can you explain the need for using the build flags for this type of stuff? It seems to me that this is SUPER error prone, because we might be testing code that's not actually going into the binary and rather just check that our testing config is working if we ever forget something to be synced between those files. 

Do we have a list of files with build tags we need to check to keep in sync? For me it's actually the first time I'm seeing this and only notice because of the duplication in this PR 👀 seems like a bad pattern but I'd be glad to understand the motivation for implementing it 🙏

Reply:
there're only 2 files that are replaced on the binary when running tests (using the `-tags=test` flag). These are:
- config_testing.go
- denom_config_testing.go

Basically what we're doing is replacing the files ('config.go' and 'denom_config.go') for the tests.
We do this to achieve the following:
- use a different pointer for the tests configurations (to avoid any weird side effect on the production binary)
- allow to reset and modify the configuration for different testing scenarios (eg. the `ResetTestChainConfig` func).